src/QMCHamiltonians/CoulombPBCAA.cpp

```diff
@@ -141,6 +141,10 @@ void CoulombPBCAA::checkout_particle_quantities(TraceManager& tm)
   streaming_particles = request.streaming_array(myName);
   if (streaming_particles)
   {
+    if (!Ps.getPerParticleSKState())
```

Comment:
If I'm following here, this is checked because certain state needs to be initialized by Ps in SK if SK being run per particle, I don't see a way to set this in SK except by calling turnOnPerParticleSK(), which makes this whole exercise seem redundant as getPerParticleSKState will only be true if turnOnPerParticleSK has already been called on SK. 

This only happens at these locations
```
src/QMCHamiltonians/CoulombPBCAA.cpp:42:    ref.turnOnPerParticleSK();
src/QMCHamiltonians/CoulombPBCAB.cpp:38:    PtclA.turnOnPerParticleSK();
src/QMCHamiltonians/EnergyDensityEstimator.cpp:69:    Pdynamic->turnOnPerParticleSK();
src/QMCHamiltonians/EnergyDensityEstimator.cpp:81:      Pstatic->turnOnPerParticleSK();
src/QMCHamiltonians/ForceChiesaPBCAA.cpp:41:  ions.turnOnPerParticleSK();
```
What is the code path that results in StructFact.StorePerParticle==true without the call to `ParticleSet::turnOnPerParticleSK`?

This strange state sequence should be explained in documentation

Reply:
I guess the check can be eliminated, and just call turnOnPerParticleSK.

Reply:
So for the other Hamiltonian components that turnOnPerParticleSK when forces are computed, do they also need to turn it on if streaming_particles == true. This would be CoulombPBCAB and ForceChiesaPBCAA

Reply:
If we had an test case with an all-electron calculation, CoulombPBCAB would probably need to be fixed.    I don't know what conditions would be needed for ForceChiesaPBCAA.    I would prefer there be a test case present before changing the code.

Reply:
I tested an all-election case, and what happened was the code in CoulombPBCAA called StorePerParticle for the electrons and also for the ions.  So when ColoumbPBCAB was called, the per-particle stuff was were already activated.
Though I could construct some unlikely cases that would make it segfault (like leaving out the ion-ion potential).

Reply:
Thanks @markdewing, can you make an issue for those segfaults.

Reply:
The remaining question is whether we drop the ParticleSet::getPerParticleSKState method.  I'd just like to drop it. Someone reading it in a year is going to wonder why its there.

Reply:
Storing SK data per particle instead of per specie needs a lot of memory. In a simulation of solide, is not a problem. In a simulation of 2D material, I have seen 3GB memory just taken by SK temporal storage which can be reduced by reduction over species. That is why the consumer is responsible to turn on the per particle setting.

Reply:
I pushed more documentation in StructFact.h